tracing.go

```diff
@@ -173,15 +167,15 @@ func StartSpan(ctx context.Context, operation string, options ...SpanOption) *Sp
 		option(&span)
 	}
 
-	span.Sampled = span.sample()
-
 	if hasParent {
 		span.recorder = parent.spanRecorder()
 	} else {
 		span.recorder = &spanRecorder{}
 	}
 	span.recorder.record(&span)
 
+	span.Sampled = span.sample()
```

Comment:
This has to happen after we record the span, since the sample method checks IsTransaction()